---
layout: lux
title: About Me
---


<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Emma Gail" />

<meta name="date" content="2019-04-08" />

<title>Mapping protein-RNA interactions between a recombinant human PRC2 to RNA using RBDmap</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>



<link rel="stylesheet" href="nsmb_pipeline_rbdmap_cayman_files/style.css" type="text/css" />

</head>

<body>




<section class="page-header">
<h1 class="title toc-ignore project-name">Mapping protein-RNA interactions between a recombinant human PRC2 to RNA using RBDmap</h1>
<h4 class="author project-author">Emma Gail</h4>
<h4 class="date project-date">2019-04-08</h4>
</section>



<section class="main-content">
<p><strong>This tutorial is still under development and may change slightly as it is being edited.</strong></p>
<div id="a-brief-introduction" class="section level1">
<h1>A Brief Introduction</h1>
<p>This tutorial is relevant to the published paper <a href="https://www.nature.com/articles/s41594-019-0197-y">RNA exploits an exposed regulatory site to inhibit the enzymatic activity of PRC2</a><a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>. However, the workflow can be used for data processed by MaxQuant.</p>
</div>
<div id="setting-up-our-environment" class="section level1">
<h1>Setting Up Our Environment</h1>
<div id="load-libraries" class="section level2">
<h2>Load Libraries</h2>
<p>First, load the libraries needed for the package and tutorial.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb1-2" title="2"><span class="kw">library</span>(bio3d)</a>
<a class="sourceLine" id="cb1-3" title="3"><span class="kw">library</span>(Biostrings)</a>
<a class="sourceLine" id="cb1-4" title="4"><span class="kw">library</span>(seqinr)</a>
<a class="sourceLine" id="cb1-5" title="5"><span class="kw">library</span>(RColorBrewer)</a>
<a class="sourceLine" id="cb1-6" title="6"><span class="kw">library</span>(openxlsx)</a>
<a class="sourceLine" id="cb1-7" title="7"><span class="kw">library</span>(viridis)</a>
<a class="sourceLine" id="cb1-8" title="8"><span class="kw">library</span>(stringr)</a>
<a class="sourceLine" id="cb1-9" title="9"><span class="kw">library</span>(svglite)</a>
<a class="sourceLine" id="cb1-10" title="10"><span class="kw">library</span>(crisscrosslinker)</a></code></pre></div>
</div>
<div id="defining-our-variables" class="section level2">
<h2>Defining Our Variables</h2>
<p>When loading the data, it is important to make sure that the files for input are named well so the functions will be able to correctly identify and organize the data.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1">sys.path &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/NSMB_RBDmap&quot;</span>,</a>
<a class="sourceLine" id="cb2-2" title="2">                        <span class="dt">package =</span> <span class="st">&#39;crisscrosslinker&#39;</span>,</a>
<a class="sourceLine" id="cb2-3" title="3">                        <span class="dt">mustWork =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-4" title="4"></a>
<a class="sourceLine" id="cb2-5" title="5"></a>
<a class="sourceLine" id="cb2-6" title="6">sys.files &lt;-<span class="st"> </span><span class="kw">list.files</span>(sys.path)</a>
<a class="sourceLine" id="cb2-7" title="7"><span class="kw">head</span>(sys.files)</a></code></pre></div>
<pre><code>## [1] &quot;Repeat01_NoXL_ArgC_eluate.txt&quot; &quot;Repeat01_NoXL_ArgC_input.txt&quot; 
## [3] &quot;Repeat01_NoXL_LysC_eluate.txt&quot; &quot;Repeat01_NoXL_LysC_input.txt&quot; 
## [5] &quot;Repeat01_UV_ArgC_eluate.txt&quot;   &quot;Repeat01_UV_ArgC_input.txt&quot;</code></pre>
<p>Each of the file names contains 3 pieces of information: the name of the experiment, if it is an input/eluate file, and which protease was used for the experiment.</p>
<p>If comparing between multiple experiments, indicate the experiment names in the code.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="co"># experiment_names &lt;- c(&#39;experiment1&#39;,&#39;experiment2&#39;)</span></a>
<a class="sourceLine" id="cb4-2" title="2">experiment_names &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;Repeat02_NoXL&#39;</span>,<span class="st">&#39;Repeat02_UV&#39;</span>)</a></code></pre></div>
<p>We will also need to load the FASTA file that was used during the initial analysis to get the tryptic peptides. For more information on how this FASTA file was generated, please refer to the paper.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1">fasta_file &lt;-<span class="st"> </span>seqinr<span class="op">::</span><span class="kw">read.fasta</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata/NSMB_FASTA&quot;</span>,</a>
<a class="sourceLine" id="cb5-2" title="2">                                             <span class="st">&#39;PRC2_5m.fasta&#39;</span>,</a>
<a class="sourceLine" id="cb5-3" title="3">                                             <span class="dt">package =</span> <span class="st">&#39;crisscrosslinker&#39;</span>,</a>
<a class="sourceLine" id="cb5-4" title="4">                                             <span class="dt">mustWork =</span> <span class="ot">TRUE</span>))</a></code></pre></div>
<p>If the file format is from MaxQuant with an Andromeda score, you will need to set it. The default setting is 20 and we will explicitly set it in this demonstration.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1">cutoff_score &lt;-<span class="st"> </span><span class="dv">20</span></a></code></pre></div>
</div>
</div>
<div id="initializing-the-data" class="section level1">
<h1>Initializing the Data</h1>
<div id="extracting-the-hits" class="section level2">
<h2>Extracting the Hits</h2>
<p>We are now ready to get a list of the relevant hits for our data.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1">sequence_hit_list &lt;-<span class="st"> </span><span class="kw">rbd.makeSeqHitList</span>(<span class="dt">fasta_file =</span> fasta_file,</a>
<a class="sourceLine" id="cb7-2" title="2">                                        <span class="dt">file_format =</span> <span class="st">&#39;txt&#39;</span>,</a>
<a class="sourceLine" id="cb7-3" title="3">                                        <span class="dt">cutoff_score =</span> cutoff_score,</a>
<a class="sourceLine" id="cb7-4" title="4">                                        <span class="dt">experiment_directory =</span> sys.path)</a></code></pre></div>
<p>Here we have found all of the tryptic peptides that:</p>
<ul>
<li>Have R/K or K/R cleavage</li>
<li>Correspond to a protein name in the FASTA file</li>
<li>Have an Andromeda score &gt; our cutoff</li>
</ul>
</div>
<div id="making-an-inputeluate-table" class="section level2">
<h2>Making an Input/Eluate Table</h2>
<p>Since we want to compare the input and eluate data from the experiments, let’s make a table comparing the two.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1">input_eluate_table &lt;-<span class="st"> </span><span class="kw">rbd.makeIETable</span>(sequence_hit_list,</a>
<a class="sourceLine" id="cb8-2" title="2">                                      experiment_names)</a></code></pre></div>
<p>We can see how the results look here:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1"><span class="kw">head</span>(input_eluate_table[,<span class="kw">c</span>(<span class="st">&#39;sequence&#39;</span>,<span class="st">&#39;input&#39;</span>,<span class="st">&#39;eluate&#39;</span>,<span class="st">&#39;protein&#39;</span>)])</a></code></pre></div>
<table>
<caption>Input/eluate table</caption>
<thead>
<tr class="header">
<th align="left">sequence</th>
<th align="right">input</th>
<th align="right">eluate</th>
<th align="left">protein</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ASMSEFLESEDGEVEQQR</td>
<td align="right">288150000</td>
<td align="right">0</td>
<td align="left">SUZ12_Q15022</td>
</tr>
<tr class="even">
<td align="left">ATWETILDGK</td>
<td align="right">2490900000</td>
<td align="right">0</td>
<td align="left">SUZ12_Q15022</td>
</tr>
<tr class="odd">
<td align="left">DEVLSADYDLLGEK</td>
<td align="right">309200000</td>
<td align="right">0</td>
<td align="left">EED_O75530</td>
</tr>
<tr class="even">
<td align="left">DPNLLLSVSK</td>
<td align="right">49414000</td>
<td align="right">0</td>
<td align="left">EED_O75530</td>
</tr>
<tr class="odd">
<td align="left">EAAFDDAVEER</td>
<td align="right">1168100000</td>
<td align="right">0</td>
<td align="left">RBBP4_Q09028</td>
</tr>
<tr class="even">
<td align="left">ECDPDLCLTCGAADHWDSK</td>
<td align="right">149440000</td>
<td align="right">0</td>
<td align="left">EZH2_Q15910-2</td>
</tr>
</tbody>
</table>
<p>We can now move on to making a plot.</p>
</div>
<div id="making-an-inputeluate-plot" class="section level2">
<h2>Making an Input/Eluate Plot</h2>
<p>As you can see, many of the hits that were found in the input were not found in the eluate. We will filter those later. For now, let’s plot everything.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1"><span class="kw">rbd.makeIEPlot</span>(input_eluate_table, experiment_names)</a></code></pre></div>
<p><img src="/Users/emmagail/Downloads/NSMB_RBDmap/ie_plot1.jpeg" /></p>
<p>We can change the default settings of the plot, including colors and the title by inputting some more variables<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1">name_of_experiment &lt;-<span class="st"> &#39;RBDmap Experiments: Input vs. Eluate&#39;</span></a>
<a class="sourceLine" id="cb11-2" title="2"><span class="kw">rbd.makeIEPlot</span>(input_eluate_table,</a>
<a class="sourceLine" id="cb11-3" title="3">               experiment_names,</a>
<a class="sourceLine" id="cb11-4" title="4">               <span class="dt">palette =</span> <span class="st">&#39;Pastel1&#39;</span>,</a>
<a class="sourceLine" id="cb11-5" title="5">               <span class="dt">fill =</span> <span class="st">&#39;white&#39;</span>,</a>
<a class="sourceLine" id="cb11-6" title="6">               <span class="dt">colour =</span> <span class="st">&#39;black&#39;</span>,</a>
<a class="sourceLine" id="cb11-7" title="7">               <span class="dt">size =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb11-8" title="8">               <span class="dt">title =</span> name_of_experiment)</a></code></pre></div>
<p><img src="/Users/emmagail/Downloads/NSMB_RBDmap/ie_plot2.jpeg" /></p>
</div>
</div>
<div id="finding-the-binding-sites" class="section level1">
<h1>Finding the Binding Sites</h1>
<p>Now that we’ve compared our control, let’s remove all results that do not show up in the eluate files.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1">filtered_input_eluate_table &lt;-<span class="st"> </span>input_eluate_table[input_eluate_table<span class="op">$</span>eluate <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>,]</a>
<a class="sourceLine" id="cb12-2" title="2">filtered_input_eluate_table</a></code></pre></div>
<table>
<caption>Filtered input/eluate table</caption>
<thead>
<tr class="header">
<th align="left">sequence</th>
<th align="right">input</th>
<th align="right">eluate</th>
<th align="left">protein</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ECSVTSDLDFPTQVIPLK</td>
<td align="right">1101400000</td>
<td align="right">13380000</td>
<td align="left">EZH2_Q15910-2</td>
</tr>
<tr class="even">
<td align="left">WLGDLILSK</td>
<td align="right">1180000000</td>
<td align="right">14292000</td>
<td align="left">EED_O75530</td>
</tr>
<tr class="odd">
<td align="left">ECSVTSDLDFPTQVIPLK</td>
<td align="right">1649500000</td>
<td align="right">16849000</td>
<td align="left">EZH2_Q15910-2</td>
</tr>
<tr class="even">
<td align="left">LQLLDGEYEVAMQEMEECPISK</td>
<td align="right">1057200000</td>
<td align="right">8763700</td>
<td align="left">SUZ12_Q15022</td>
</tr>
<tr class="odd">
<td align="left">WLGDLILSK</td>
<td align="right">1288100000</td>
<td align="right">22946000</td>
<td align="left">EED_O75530</td>
</tr>
</tbody>
</table>
<p>We’ll use these results for finding binding sequences. We’ll first get a data.frame of the binding sequences aligned to the FASTA file.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1">bs_output_fasta &lt;-<span class="st"> </span><span class="kw">rbd.getBSfromIET</span>(filtered_input_eluate_table,</a>
<a class="sourceLine" id="cb13-2" title="2">                                    <span class="dt">fasta_file =</span> fasta_file,</a>
<a class="sourceLine" id="cb13-3" title="3">                                    <span class="dt">align_to =</span> <span class="st">&#39;FASTA&#39;</span>)</a>
<a class="sourceLine" id="cb13-4" title="4"></a>
<a class="sourceLine" id="cb13-5" title="5">bs_output_fasta[<span class="kw">c</span>(<span class="st">&#39;pro_name&#39;</span>,<span class="st">&#39;binding_sequence&#39;</span>)]</a></code></pre></div>
<table>
<caption>Detected binding sequences from FASTA</caption>
<thead>
<tr class="header">
<th align="left">pro_name</th>
<th align="left">binding_sequence</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">EZH2_Q15910-2</td>
<td align="left">TLNAVASVPIMYSWSPLQQNFMVEDETVLHNIPYMGDEVLDQDGTFIEELIKNYDGKVHGDR</td>
</tr>
<tr class="even">
<td align="left">EED_O75530</td>
<td align="left">SCENAIVCWKPGKMEDDIDKIKPSESNVTILGR</td>
</tr>
<tr class="odd">
<td align="left">EZH2_Q15910-2</td>
<td align="left">QRRIQPVHILTSVSSLRGTR</td>
</tr>
<tr class="even">
<td align="left">SUZ12_Q15022</td>
<td align="left">NRR</td>
</tr>
<tr class="odd">
<td align="left">EED_O75530</td>
<td align="left">IHFPDFSTRDIHRNYVDCVR</td>
</tr>
</tbody>
</table>
<div id="making-a-heatmap" class="section level2">
<h2>Making a Heatmap</h2>
<p>Let’s make a heatmap and sort the sequences by their names.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1">bs_freqVector &lt;-<span class="st"> </span><span class="kw">rbd.freqVector</span>(<span class="dt">bs_output =</span> bs_output_fasta,</a>
<a class="sourceLine" id="cb14-2" title="2">                                <span class="dt">name_by =</span> <span class="st">&#39;pro_name&#39;</span>,</a>
<a class="sourceLine" id="cb14-3" title="3">                                <span class="dt">heatmap =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb14-4" title="4">                                <span class="dt">db_selection =</span> <span class="st">&#39;FASTA&#39;</span>)</a></code></pre></div>
<pre><code>## Warning: package &#39;gdtools&#39; was built under R version 3.5.2</code></pre>
<p><img src="nsmb_pipeline_rbdmap_cayman_files/figure-html/unnamed-chunk-11-1.png" /><!-- --></p>
<p>We can also adjust the colors of the heatmap if we so wish. You can put in a standard <code>RColorBrewer</code> or <code>viridis</code> palette or a list of hexcodes and/or standard R Colors.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" title="1"><span class="kw">rbd.freqVector</span>(<span class="dt">bs_output =</span> bs_output_fasta,</a>
<a class="sourceLine" id="cb16-2" title="2">               <span class="dt">name_by =</span> <span class="st">&#39;pro_name&#39;</span>,</a>
<a class="sourceLine" id="cb16-3" title="3">               <span class="dt">heatmap =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb16-4" title="4">               <span class="dt">db_selection =</span> <span class="st">&#39;FASTA&#39;</span>,</a>
<a class="sourceLine" id="cb16-5" title="5">               <span class="dt">colors =</span> <span class="st">&quot;Blues&quot;</span>,</a>
<a class="sourceLine" id="cb16-6" title="6">               <span class="dt">save_plot =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p><img src="nsmb_pipeline_rbdmap_cayman_files/figure-html/unnamed-chunk-12-1.png" /><!-- --></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" title="1"><span class="kw">rbd.freqVector</span>(<span class="dt">bs_output =</span> bs_output_fasta,</a>
<a class="sourceLine" id="cb17-2" title="2">               <span class="dt">name_by =</span> <span class="st">&#39;pro_name&#39;</span>,</a>
<a class="sourceLine" id="cb17-3" title="3">               <span class="dt">heatmap =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb17-4" title="4">               <span class="dt">db_selection =</span> <span class="st">&#39;FASTA&#39;</span>,</a>
<a class="sourceLine" id="cb17-5" title="5">               <span class="dt">colors =</span> <span class="kw">c</span>(<span class="st">&#39;#b3ecec&#39;</span>,<span class="st">&#39;#43e8d8&#39;</span>,<span class="st">&#39;#48d1cc&#39;</span>),</a>
<a class="sourceLine" id="cb17-6" title="6">               <span class="dt">save_plot =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p><img src="nsmb_pipeline_rbdmap_cayman_files/figure-html/unnamed-chunk-13-1.png" /><!-- --></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" title="1"><span class="kw">rbd.freqVector</span>(<span class="dt">bs_output =</span> bs_output_fasta,</a>
<a class="sourceLine" id="cb18-2" title="2">               <span class="dt">name_by =</span> <span class="st">&#39;pro_name&#39;</span>,</a>
<a class="sourceLine" id="cb18-3" title="3">               <span class="dt">heatmap =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb18-4" title="4">               <span class="dt">db_selection =</span> <span class="st">&#39;FASTA&#39;</span>,</a>
<a class="sourceLine" id="cb18-5" title="5">               <span class="co"># palette from color-hex.com</span></a>
<a class="sourceLine" id="cb18-6" title="6">               <span class="dt">colors =</span> <span class="kw">c</span>(<span class="st">&quot;#465ca8&quot;</span>,<span class="st">&quot;#7767b8&quot;</span>,<span class="st">&quot;#a272c4&quot;</span>,</a>
<a class="sourceLine" id="cb18-7" title="7">                           <span class="st">&quot;#cc7ecc&quot;</span>,<span class="st">&#39;#f38bd1&#39;</span>), </a>
<a class="sourceLine" id="cb18-8" title="8">               <span class="dt">save_plot =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p><img src="nsmb_pipeline_rbdmap_cayman_files/figure-html/unnamed-chunk-14-1.png" /><!-- --></p>
<p>If you require further customization, we recommend saving the output from <code>rbd.freqVector</code> and using that as input for a heatmap.</p>
</div>
<div id="aligning-peptides-to-a-pdb-file" class="section level2">
<h2>Aligning Peptides to a PDB File</h2>
<p>We will now align the binding sequences to a PDB file. For simplicity, we’ll only focus on one protein: EZH2.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" title="1">ezh2_iet &lt;-<span class="st"> </span>filtered_input_eluate_table[filtered_input_eluate_table<span class="op">$</span>protein <span class="op">==</span><span class="st"> &#39;EZH2_Q15910-2&#39;</span>,]</a></code></pre></div>
<p>We already know the PDB file and chain that we want to align the proteins to. In this case, we can create and add a new table:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" title="1">protein_dict &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata/NSMB_misc&quot;</span>,</a>
<a class="sourceLine" id="cb20-2" title="2">                                     <span class="st">&#39;PRC2_protein_dict.csv&#39;</span>,</a>
<a class="sourceLine" id="cb20-3" title="3">                                     <span class="dt">package =</span> <span class="st">&#39;crisscrosslinker&#39;</span>,</a>
<a class="sourceLine" id="cb20-4" title="4">                                     <span class="dt">mustWork =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb20-5" title="5">protein_dict</a></code></pre></div>
<table>
<caption>Protein dictionary with corresponding UniProt and PDB identifiers</caption>
<thead>
<tr class="header">
<th align="left">protein_id</th>
<th align="left">uniprot_id</th>
<th align="left">pdb_id</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">EZH2_Q15910-2</td>
<td align="left">Q15910-2</td>
<td align="left">6C23_C</td>
</tr>
<tr class="even">
<td align="left">EED_O75530</td>
<td align="left">O75530</td>
<td align="left">6C23_L</td>
</tr>
<tr class="odd">
<td align="left">RBBP4_Q09028</td>
<td align="left">Q09028</td>
<td align="left">6C23_N</td>
</tr>
<tr class="even">
<td align="left">SUZ12_Q15022</td>
<td align="left">Q15022</td>
<td align="left">6C23_A</td>
</tr>
<tr class="odd">
<td align="left">AEBP2_Q6ZN18</td>
<td align="left">Q6ZN18-2</td>
<td align="left">6C23_P</td>
</tr>
</tbody>
</table>
<p>Now we don’t have to go through menus to be able to define the PDB ID/chain that we need to use. This will save us time when making our table of binding sequences.</p>
<p>You can always leave the dictionary blank in order to access an interactive menu to be able to use BLAST in order to align your sequence to a known PDB structure.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" title="1">bs_ezh2 &lt;-<span class="st"> </span><span class="kw">rbd.getBSfromIET</span>(ezh2_iet,</a>
<a class="sourceLine" id="cb21-2" title="2">                            <span class="dt">fasta_file =</span> fasta_file,</a>
<a class="sourceLine" id="cb21-3" title="3">                            <span class="dt">align_to =</span> <span class="st">&#39;PDB&#39;</span>,</a>
<a class="sourceLine" id="cb21-4" title="4">                            <span class="dt">protein_dict =</span> protein_dict)</a></code></pre></div>
<p>Let’s see how many matches we had to the chain we chose:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" title="1"><span class="kw">table</span>(bs_ezh2<span class="op">$</span>db)</a></code></pre></div>
<table>
<caption>EZH2 PDB Matches</caption>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PDB</td>
<td align="right">2</td>
</tr>
</tbody>
</table>
<p>2 matched our chain! Let’s visualize it.</p>
</div>
<div id="making-the-pymol-file" class="section level2">
<h2>Making the PyMOL File</h2>
<p>Now that we have the binding sequences aligned to a PDB file, we can create a PyMOL file.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" title="1"><span class="kw">rbd.pymol</span>(bs_ezh2, </a>
<a class="sourceLine" id="cb23-2" title="2">          <span class="dt">color_by =</span> <span class="st">&#39;binding_sequence&#39;</span>,</a>
<a class="sourceLine" id="cb23-3" title="3">          <span class="dt">colors =</span> <span class="ot">NULL</span>) <span class="co">#default colors</span></a></code></pre></div>
<p><img src="nsmb_pipeline_rbdmap_cayman_files/figure-html/unnamed-chunk-20-1.png" /><!-- --></p>
<p><img src="/Users/emmagail/Downloads/NSMB_RBDmap/pymol_generic_colors.png" style="width:100.0%" /></p>
<p>There is an additional legend with the colors corresponding to the binding sequences. The colors will be assigned based on the order the binding sequences are in. If you would like the sequences in a different order, please order them before executing rbd.pymol().</p>
<p>If we do not choose any colors, it will default to the PyMOL tints palette. However, we can input any RColorBrewer or viridis palette as well as standard PyMOL/R colors or hexcodes.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" title="1"><span class="kw">rbd.pymol</span>(bs_ezh2, </a>
<a class="sourceLine" id="cb24-2" title="2">          <span class="dt">color_by =</span> <span class="st">&#39;binding_sequence&#39;</span>, </a>
<a class="sourceLine" id="cb24-3" title="3">          <span class="dt">colors =</span> <span class="st">&#39;Dark2&#39;</span>, </a>
<a class="sourceLine" id="cb24-4" title="4">          <span class="dt">file.name =</span> <span class="st">&#39;pymol_Dark2.pml&#39;</span>) <span class="co"># RColorBrewer palette</span></a></code></pre></div>
<p><img src="nsmb_pipeline_rbdmap_cayman_files/figure-html/unnamed-chunk-21-1.png" /><!-- --></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" title="1"><span class="kw">rbd.pymol</span>(bs_ezh2, </a>
<a class="sourceLine" id="cb25-2" title="2">          <span class="dt">color_by =</span> <span class="st">&#39;binding_sequence&#39;</span>, </a>
<a class="sourceLine" id="cb25-3" title="3">          <span class="dt">colors =</span> <span class="st">&#39;magma&#39;</span>, </a>
<a class="sourceLine" id="cb25-4" title="4">          <span class="dt">file.name =</span> <span class="st">&#39;pymol_magma.pml&#39;</span>) <span class="co"># viridis palette</span></a></code></pre></div>
<p><img src="nsmb_pipeline_rbdmap_cayman_files/figure-html/unnamed-chunk-21-2.png" /><!-- --></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" title="1"><span class="kw">rbd.pymol</span>(bs_ezh2, </a>
<a class="sourceLine" id="cb26-2" title="2">          <span class="dt">color_by =</span> <span class="st">&#39;binding_sequence&#39;</span>, </a>
<a class="sourceLine" id="cb26-3" title="3">          <span class="dt">colors =</span> <span class="kw">c</span>(<span class="st">&#39;red&#39;</span>,<span class="st">&#39;blue&#39;</span>), </a>
<a class="sourceLine" id="cb26-4" title="4">          <span class="dt">file.name =</span> <span class="st">&#39;pymol_rgb.pml&#39;</span>) <span class="co"># standard colors</span></a></code></pre></div>
<p><img src="nsmb_pipeline_rbdmap_cayman_files/figure-html/unnamed-chunk-21-3.png" /><!-- --></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" title="1"><span class="kw">rbd.pymol</span>(bs_ezh2, </a>
<a class="sourceLine" id="cb27-2" title="2">          <span class="dt">color_by =</span> <span class="st">&#39;binding_sequence&#39;</span>, </a>
<a class="sourceLine" id="cb27-3" title="3">          <span class="dt">colors =</span> <span class="kw">c</span>(<span class="st">&quot;#f8a1be&quot;</span>,<span class="st">&quot;#ffb6b1&quot;</span>), </a>
<a class="sourceLine" id="cb27-4" title="4">          <span class="dt">file.name =</span> <span class="st">&#39;pymol_hexcodes.pml&#39;</span>) <span class="co"># hexcodes</span></a></code></pre></div>
<p><img src="nsmb_pipeline_rbdmap_cayman_files/figure-html/unnamed-chunk-21-4.png" /><!-- --></p>

</div>
</div>
<div id="differential-analysis" class="section level1">
<h1>Differential Analysis</h1>
<p>We have now looked at one experiment, but now we want to look at multiple experiments we have run and compare between them. First we will do what we have done before, but combine the different runs together to examine how often these binding sites appear<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>.</p>
<p>Each of these experiments have slighly different prefixes, so we will adjust them accordingly with a list. In this instance, we could probably get away with just using “UV” for all 3 repeats. However, for the purposes of this tutorial, we will explicitly define the prefixes.</p>
<p>We’re now ready to proceed to get the binding sequences for each of these repeats.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" title="1">bs_output_diff_analysis &lt;-<span class="st"> </span><span class="kw">data.frame</span>()</a>
<a class="sourceLine" id="cb28-2" title="2">num_repeats &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">4</span></a>
<a class="sourceLine" id="cb28-3" title="3">repeat_names &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&#39;Repeat0&#39;</span>,num_repeats)</a>
<a class="sourceLine" id="cb28-4" title="4"></a>
<a class="sourceLine" id="cb28-5" title="5"><span class="cf">for</span>(rname <span class="cf">in</span> repeat_names){</a>
<a class="sourceLine" id="cb28-6" title="6">  shl &lt;-<span class="st"> </span>sequence_hit_list[<span class="kw">startsWith</span>(<span class="kw">names</span>(sequence_hit_list),rname)]</a>
<a class="sourceLine" id="cb28-7" title="7">  enames &lt;-<span class="st"> </span><span class="kw">paste0</span>(rname,<span class="st">&#39;_&#39;</span>,<span class="kw">c</span>(<span class="st">&#39;UV&#39;</span>,<span class="st">&#39;NoXL&#39;</span>))</a>
<a class="sourceLine" id="cb28-8" title="8">  input_eluate_table &lt;-<span class="st"> </span><span class="kw">rbd.makeIETable</span>(shl, enames)</a>
<a class="sourceLine" id="cb28-9" title="9">  filtered_input_eluate_table &lt;-<span class="st"> </span>input_eluate_table[input_eluate_table<span class="op">$</span>eluate <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>,]</a>
<a class="sourceLine" id="cb28-10" title="10">  </a>
<a class="sourceLine" id="cb28-11" title="11">  bs_output_pdb &lt;-<span class="st"> </span><span class="kw">rbd.getBSfromIET</span>(filtered_input_eluate_table,</a>
<a class="sourceLine" id="cb28-12" title="12">                                  <span class="dt">fasta_file =</span> fasta_file,</a>
<a class="sourceLine" id="cb28-13" title="13">                                  <span class="dt">align_to =</span> <span class="st">&#39;PDB&#39;</span>,</a>
<a class="sourceLine" id="cb28-14" title="14">                                  <span class="dt">protein_dict =</span> protein_dict)</a>
<a class="sourceLine" id="cb28-15" title="15">  </a>
<a class="sourceLine" id="cb28-16" title="16">  bs_output_diff_analysis &lt;-<span class="st"> </span><span class="kw">rbind</span>(bs_output_diff_analysis,<span class="kw">data.frame</span>(bs_output_pdb))</a>
<a class="sourceLine" id="cb28-17" title="17">}</a></code></pre></div>
<p>Now that we have loaded the data into our R environment, let’s make a PyMOL file<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" title="1"><span class="kw">rbd.pymol</span>(bs_output_diff_analysis, </a>
<a class="sourceLine" id="cb29-2" title="2">          <span class="dt">color_by =</span> <span class="st">&#39;freq&#39;</span>, </a>
<a class="sourceLine" id="cb29-3" title="3">          <span class="dt">colors =</span> <span class="st">&#39;magma&#39;</span>, </a>
<a class="sourceLine" id="cb29-4" title="4">          <span class="dt">file.name =</span> <span class="st">&#39;pymol_magma_diff_analysis.pml&#39;</span>)</a></code></pre></div>
<p><img src="nsmb_pipeline_rbdmap_cayman_files/figure-html/unnamed-chunk-22-1.png" /><!-- --></p>
<p><img src="/Users/emmagail/Downloads/NSMB_RBDmap/pymol_magma_diff_analysis.png" style="width:50.0%" /> <img src="/Users/emmagail/Downloads/NSMB_RBDmap/freqvector_legend_test2.png" style="width:20.0%" /></p>
</div>
<div id="session-info" class="section level1">
<h1>Session Info</h1>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" title="1">devtools<span class="op">::</span><span class="kw">session_info</span>()</a></code></pre></div>
<pre><code>## ─ Session info ──────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 3.5.1 (2018-07-02)
##  os       macOS  10.14.3              
##  system   x86_64, darwin15.6.0        
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       Australia/Melbourne         
##  date     2019-04-08                  
## 
## ─ Packages ──────────────────────────────────────────────────────────────
##  package          * version    date       lib
##  ade4               1.7-13     2018-08-31 [1]
##  assertthat         0.2.1      2019-03-21 [1]
##  backports          1.1.3      2018-12-14 [1]
##  bio3d            * 2.3-4      2018-04-03 [1]
##  BiocGenerics     * 0.28.0     2018-10-30 [1]
##  Biostrings       * 2.50.2     2019-01-03 [1]
##  callr              3.2.0      2019-03-15 [1]
##  cli                1.1.0      2019-03-19 [1]
##  colorspace         1.4-1      2019-03-18 [1]
##  crayon             1.3.4      2017-09-16 [1]
##  crisscrosslinker * 0.9        2019-04-02 [1]
##  desc               1.2.0      2018-05-01 [1]
##  devtools           2.0.1      2018-10-26 [1]
##  digest             0.6.18     2018-10-10 [1]
##  dplyr              0.8.0.1    2019-02-15 [1]
##  evaluate           0.13       2019-02-12 [1]
##  fs                 1.2.7      2019-03-19 [1]
##  gdtools          * 0.1.8      2019-04-02 [1]
##  ggplot2          * 3.1.0      2018-10-25 [1]
##  glue               1.3.1      2019-03-12 [1]
##  gridExtra          2.3        2017-09-09 [1]
##  gtable             0.3.0      2019-03-25 [1]
##  highr              0.8        2019-03-20 [1]
##  htmltools          0.3.6      2017-04-28 [1]
##  httr               1.4.0      2018-12-11 [1]
##  IRanges          * 2.16.0     2018-10-30 [1]
##  jsonlite           1.6        2018-12-07 [1]
##  knitr              1.22       2019-03-08 [1]
##  labeling           0.3        2014-08-23 [1]
##  lattice            0.20-38    2018-11-04 [1]
##  lava               1.6.5      2019-02-12 [1]
##  lazyeval           0.2.2      2019-03-15 [1]
##  magrittr           1.5        2014-11-22 [1]
##  MASS               7.3-51.4   2019-03-31 [1]
##  Matrix             1.2-17     2019-03-22 [1]
##  memoise            1.1.0      2017-04-21 [1]
##  munsell            0.5.0      2018-06-12 [1]
##  openxlsx         * 4.1.0      2018-05-26 [1]
##  pillar             1.3.1      2018-12-15 [1]
##  pkgbuild           1.0.3      2019-03-20 [1]
##  pkgconfig          2.0.2      2018-08-16 [1]
##  pkgload            1.0.2      2018-10-29 [1]
##  plyr               1.8.4      2016-06-08 [1]
##  prettydoc          0.2.1      2018-01-16 [1]
##  prettyunits        1.0.2      2015-07-13 [1]
##  processx           3.3.0      2019-03-10 [1]
##  prodlim            2018.04.18 2018-04-18 [1]
##  ps                 1.3.0      2018-12-21 [1]
##  purrr              0.3.2      2019-03-15 [1]
##  R6                 2.4.0      2019-02-14 [1]
##  RColorBrewer     * 1.1-2      2014-12-07 [1]
##  Rcpp               1.0.1      2019-03-17 [1]
##  remotes            2.0.2      2018-10-30 [1]
##  rlang              0.3.1      2019-01-08 [1]
##  rmarkdown          1.12       2019-03-14 [1]
##  rprojroot          1.3-2      2018-01-03 [1]
##  S4Vectors        * 0.20.1     2018-11-09 [1]
##  scales             1.0.0      2018-08-09 [1]
##  seqinr           * 3.4-5      2017-08-01 [1]
##  sessioninfo        1.1.1      2018-11-05 [1]
##  stringi            1.4.3      2019-03-12 [1]
##  stringr          * 1.4.0      2019-02-10 [1]
##  survival           2.44-1.1   2019-04-01 [1]
##  svglite          * 1.2.1      2017-09-11 [1]
##  tibble             2.1.1      2019-03-16 [1]
##  tidyselect         0.2.5      2018-10-11 [1]
##  usethis            1.4.0      2018-08-14 [1]
##  viridis          * 0.5.1      2018-03-29 [1]
##  viridisLite      * 0.3.0      2018-02-01 [1]
##  withr              2.1.2      2018-03-15 [1]
##  xfun               0.6        2019-04-02 [1]
##  xml2               1.2.0      2018-01-24 [1]
##  XVector          * 0.22.0     2018-10-30 [1]
##  yaml               2.2.0      2018-07-25 [1]
##  zip                2.0.1      2019-03-11 [1]
##  zlibbioc           1.28.0     2018-10-30 [1]
##  source                                   
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  Bioconductor                             
##  Bioconductor                             
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  Github (egmg726/crisscrosslinker@2d0bd67)
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.1)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  Bioconductor                             
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  Bioconductor                             
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  CRAN (R 3.5.0)                           
##  Bioconductor                             
##  CRAN (R 3.5.0)                           
##  CRAN (R 3.5.2)                           
##  Bioconductor                             
## 
## [1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library</code></pre>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<ol style="list-style-type: decimal">
<li><p>Castello, A. et al. Comprehensive Identification of RNA-Binding Domains in Human Cells. Molecular Cell 63, 696-710, <a href="doi:10.1016/j.molcel.2016.06.029" class="uri">doi:10.1016/j.molcel.2016.06.029</a> (2016).</p></li>
<li><p>Tyanova, S., Temu, T. &amp; Cox, J. The MaxQuant computational platform for mass spectrometry-based shotgun proteomics. Nature Protocols 11, 2301-2319, <a href="doi:10.1038/nprot.2016.136" class="uri">doi:10.1038/nprot.2016.136</a> (2016).</p></li>
</ol>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>The data for this tutorial is available <a href="https://doi.org/10.26180/5c3d9751c64ae">here</a>.The relevant files are located within the .zip files for each repeat as MaxQuant_txt_folder_output/peptides.txt. For simplicity, all “peptides.txt” files have been re-named according to their experiment numbers as part of the external data available in <code>crisscrosslinkeR</code>.<a href="#fnref1" class="footnote-back">↩</a></p></li>
<li id="fn2"><p>These variables correspond to the <code>ggplot2</code> input variables that is used to generate this plot.<a href="#fnref2" class="footnote-back">↩</a></p></li>
<li id="fn3"><p>Since slightly different but overlapping peptides may show up due to the use of different proteases, the frequency is measured by the amino acid rather than the full peptide.<a href="#fnref3" class="footnote-back">↩</a></p></li>
<li id="fn4"><p>By default, it will also create a heatmap with the same color scheme as the PyMOL file if <code>heatmap</code> is kept as <code>TRUE</code>. A legend will also be generated to go with the PyMOL file, though it will be similar to the one generated by the heatmap.<a href="#fnref4" class="footnote-back">↩</a></p></li>
</ol>
</div>
</section>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
