---
layout: default
title: crisscrosslinker
---

<div class="jumbotron">
  <h1 class="display-4">crisscrosslinkeR</h1>
  <p class="lead">A package for analyzing protein-protein and protein-RNA interactions.</p>
  <hr class="my-4">
  <p></p>
</div>


<div class="row" style="padding-left:30px">
	<h1>Introduction</h1> <br><br>
	</div>
	<div class="row" style="padding-left:30px">
	<p><code>crisscrosslinkeR</code> helps to optimize XL-MS and RBDmap analysis to manage a large number of repeats with easy-to-use functions in R.</p>

	

	<p>In conjunction with visualization tools such as xiNET and PyMOL, <code>crisscrosslinkeR</code> can help 
	to streamline your ability to create intuitive, colorful, and informative graphics suitable for publication.</p>


	
	<p>Read more about the data generated by the lab that was used in some of the tutorials <a href="https://doi.org/10.1038/s41594-019-0197-y">here</a>.</p>
	
	</div>
	
	
	<div class="row" style="padding-left:30px">
	<h1>Features</h1> <br><br>
	</div>

	
	<div class="row" style="padding-left:30px">
	
	<p> crisscrosslinkeR contains a number of features to help ease data analysis of XL-MS and RBDmap datasets 
	for reproducibility and preparation of tables and figures for publication.</p>
	<br>

 <h2> Data reduction and filtration </h2>
<p>Data reduction is done across multiple samples, allowing to either merge replicates or 
accept data points based on user-defined conditions (e.g. only under the condition the same 
crosslinked peptide appeared in a given number of replicates or samples). 
Filtration is based on signifi-cance scores that were assigned during the primary data analysis.
 crisscrosslinkeR is designed to handle an unlimited number of XL-MS and RBDmap datasets simultaneously 
 for a direct comparison or data reduction, based on user preference.</p> 
<h2>Customized options for export and downstream visualization</h2>
 <p>There are currently no tools to visualize RBDmap data over sequences (2D visualization) 
 or structures (3D visualization) of protein complex-es, albeit these forms of visualization
  are popular and supported in case of XL-MS (Courcelles, et al., 2017; Holding, et al., 2013; 
  Kosinski, et al., 2015; Riffle, et al., 2016). crisscrosslinkeR takes advantage of the
  color palettes available in R, as well as custom palettes, to illustrate replicates or 
  different experimental treatments for 2D representation of RBDmap data. For 2D visualization of 
  XL-MS and, optionally, RBDmap data, crisscrosslinkeR exports files compatible with the user-friendly 
  server xiView (Graham, et al., 2019). For 3D visualization of RBDmap and XL-MS data, 
  crisscrosslinkeR uses the RCSB PDB mapping API to identify the most suitable 3D struc-ture(s) in the PDB,
   aligns it to the sequence of the protein used during the experiment and then use the selected structure 
   for 3D visualization of either RBDmap or XL-MS data. Specifically, crisscrosslinkeR
    generates a Python script to lunch PyMOL (Schroding-er, 2015) — a popular and stable suite for molecular
     imaging — which the user can then use to generate publication-quality figures or for structural analysis.
      crisscrosslinkR can be used to hide or present color-coded crosslinking sites based on the score applied
       by the primary software that detected them (MaxQuant (Tyanova, et al., 2016) or pLink (Yang, et al., 2012)),
       the number of times linked-peptides were detected across multiple replicates, differential crosslinked sites 
       (e.g. linked-peptides that appeared only under a certain treatment) and, when applicable, the crosslinking distances; 
       all these are visualized with large flexibility for user-defined color selection.</p>

<p>Collectively, crisscrosslinkeR is a versatile pipeline that streamlines the workflow,
 starting from multiple raw data files down to publication-quality figures.</p>	
	
	</div>
	
	
<hr class="my-4">	
	
	<div class="row" style="padding-left:30px">
	<h2>Installation</h2><br><br></div>
		<div class="row" style="padding-left:30px">
	<p>The package is currently available for download via GitHub. If you are unfamiliar with installing directly from GitHub, please enter the following commands in your RStudio console:</p>
	</div>
	
	
	<div class="row" style="padding-left:30px">
	<code> library("devtools")<br> install_github("egmg726/crisscrosslinker")</code>
	</div>
	
<hr class="my-4">

	<div class="row" style="padding-left:30px">
	<h2>Documentation</h2><br><br></div>
	
	<div class="row" style="padding-left:30px">
	<p> You can read the full documentation <a href="https://drive.google.com/file/d/1cxJQGFxDAbs2XezuMWhD9CwHxAd617hj/view?usp=sharing">here</a>.
	However, we recommend going through the tutorials/workflows first. <b>Please note that the documentation is currently under construction and will updated soon.</b> </p>
	</div>
	
	<div class="row" style="align-items: center;justify-content: center;">
	<iframe src="https://drive.google.com/file/d/1cxJQGFxDAbs2XezuMWhD9CwHxAd617hj/preview" align='middle' style="float:middle;" height='480' width='640'></iframe>
	</div>

<hr class="my-4">

<div class="row">
<h1 style="padding-bottom:10px;padding-left:30px">Workflows</h1>
<br> <br>
</div>

<div class="row" style="padding-top:10px">
<div class="card-deck mb-2 mx-auto">
<div class="card border-secondary mb-2" style="max-width:30rem;">
  <div class="card-header"><b>Protein-Protein Interactions</b></div>
  <div class="card-body">
    <h4 class="card-title">Features</h4>
    <div style='padding-left:5px'>
    <li>Data reduction of XL-MS repeats</li>
    <li>BLAST integration and interactive menus within R</li>
    <li>Alignment to UniProt and PDB files</li>
    <li>Export for xiNET and PyMOL visualizations</li>
    </div>
    <hr class="my-4">
        <h4 class="card-title">Tutorials</h4>
            <div style='padding-left:5px'>
    <p><a href="https://drive.google.com/file/d/1KGkrrhnsQd7tE5-L-VqbLl8AKcNoASoT/view?usp=sharing">Visual Guide </a></p>
        <p><a href="yeast_ppi_workflow.html">Identification of protein-protein interactions in the RNA polymerase II-TFIIF complex (Chen et al 2009), including 2D and 3D visualization, starting from XL-MS data </a></p>
    <p><a href="nsmb_pipeline_ppi.html">Identification of protein-protein interactions between PRC2 to its accessory subunits AEBP2, PHF19, MTF2 and EPOP (NSMB 2019)</a></p>
    </div>
  </div>
</div>
<div class="card border-secondary mb-2" style="max-width:30rem;">
  <div class="card-header"><b>Protein-RNA Interactions</b></div>
  <div class="card-body">
      <h4 class="card-title">Features</h4>
    <div style='padding-left:5px'>
    <li>Proteolytic fragment detection of RBDmap experiments</li>
    <li>Frequency analysis of RBD-associated amino acids</li>
    <li>Heatmaps of frequency analysis within R</li>
    <li>Export for PyMOL visualizations</li>
    </div>
<hr class="my-4">
    <h4 class="card-title">Tutorials</h4>
    <div style='padding-left:5px'>
    <p><a href="https://drive.google.com/file/d/1rIGt8IKLd3meyypQ07xPf_Fe_mmA302x/view?usp=sharing">Visual Guide </a></p>
    <p><a href="nsmb_pipeline_rbdmap.html">Mapping protein-RNA interactions between a recombinant human PRC2 to RNA using RBDmap (NSMB 2019)</a></p>
    <p><a href='rbdmap_prkdc_workflow.html'>Mapping the RNA binding site in PRKDC protein from RBDmap done in cells (data from Castello et al. 2016)</a></p>
    <p><a href="nsmb_rbdmap_xlms_combo.html">Combining protein-protein and protein-RNA interaction data from PRC2 into 2D and 3D visualizations</a></p>
    </div>
    
  </div>
</div>
</div> <!--card-deck mb-3 text-center"-->
</div> <!--row-->

<hr class="my-4">

<div class="row">
<h1 style="padding-bottom:10px;padding-left:30px">Gallery</h1>
<br> <br>
</div>

<div class="row" style="align-items:center;justify-content:center;"">

<div id="carouselExampleIndicators" class="carousel slide" data-ride="carousel">
  <ol class="carousel-indicators">
    <li data-target="#carouselExampleIndicators" data-slide-to="0" class="active"></li>
    <li data-target="#carouselExampleIndicators" data-slide-to="1"></li>
    <li data-target="#carouselExampleIndicators" data-slide-to="2"></li>
  </ol>
  <div class="carousel-inner" style="max-width:600x!important;max-height:600px!important;">
    <div class="carousel-item active">
      <img src="https://drive.google.com/uc?export=view&id=1Px_BXqn9ln7LTNW8azRVUw8Zk8kPSZKZ" alt="First slide" width="580px" height="410px">
    </div>
    <div class="carousel-item">
      <img src="https://drive.google.com/uc?export=view&id=1oVggGzb0T7hrfh0J9gIkdi9CN0ZtcOs3" alt="Second slide" width="600px" height="300px">
    </div>
    <div class="carousel-item">
      <img src="https://drive.google.com/uc?export=view&id=1Px_BXqn9ln7LTNW8azRVUw8Zk8kPSZKZ" alt="Third slide" width="580px" height="410px">
    </div>
  </div>
  <a class="carousel-control-prev" href="#carouselExampleIndicators" role="button" data-slide="prev">
    <span class="carousel-control-prev-icon" aria-hidden="true"></span>
    <span class="sr-only">Previous</span>
  </a>
  <a class="carousel-control-next" href="#carouselExampleIndicators" role="button" data-slide="next">
    <span class="carousel-control-next-icon" aria-hidden="true"></span>
    <span class="sr-only">Next</span>
  </a>
</div>

</div>

<hr class="my-4">
	<div class="row" style="padding-left:30px">
	<h1>References</h1> <br><br>
	</div>
	<div class="row" style="padding-left:30px">
	
<p>Castello, Alfredo, et al. (2016), 'Comprehensive Identification of RNA-Binding Domains in Human Cells', Molecular Cell, 63 (4), 696-710.</p>
<p>Castello, Alfredo, et al. (2017), 'Identification of RNA-binding domains of RNA-binding proteins in cultured cells on a system-wide scale with RBDmap', Nature Protocols, 12 (12), 2447-64.</p>
<p>Courcelles, M., et al. (2017), 'CLMSVault: A Software Suite for Protein Cross-Linking Mass-Spectrometry Data Analysis and Visualization', J Proteome Res, 16 (7), 2645-52.</p>
<p>Cox, Jürgen, et al. (2011), 'Andromeda: A peptide search engine integrated into the MaxQuant environment', Journal of Proteome Research, 10 (4), 1794-805.</p>
<p>Graham, Martin J., et al. (2019), 'xiView: A common platform for the down-stream analysis of Crosslinking Mass Spectrometry data', bioRxiv, 561829.</p>
<p>Grant, B. J., et al. (2006), 'Bio3d: an R package for the comparative analysis of protein structures', Bioinformatics, 22 (21), 2695-96.</p>
<p>Holding, A. N. (2015), 'XL-MS: Protein cross-linking coupled with mass spectrometry', Methods, 89, 54-63.</p>
<p>Holding, A. N., et al. (2013), 'Hekate: software suite for the mass spectrometric analysis and three-dimensional visualization of cross-linked protein sam-ples', J Proteome Res, 12 (12), 5923-33.</p>
<p>Kosinski, Jan, et al. (2015), 'Xlink analyzer: Software for analysis and visuali-zation of cross-linking data in the context of three-dimensional structures', Journal of Structural Biology, 189 (3), 177-83.</p>
<p>Leitner, A., et al. (2016), 'Crosslinking and Mass Spectrometry: An Integrated Technology to Understand the Structure and Function of Molecular Ma-chines', Trends Biochem Sci, 41 (1), 20-32.</p>
<p>Riffle, Michael, et al. (2016), 'ProXL (Protein Cross-Linking Database): A Platform for Analysis, Visualization, and Sharing of Protein Cross-Linking Mass Spectrometry Data', Journal of proteome research, 15 (8), 2863-70.</p>
<p>Schmidt, Carla, Kramer, Katharina, and Urlaub, Henning (2012), 'Investigation of protein–RNA interactions by mass spectrometry—Techniques and appli-cations', Journal of Proteomics, 75 (12), 3478-94.</p>
<p>Schrodinger, LLC (2015), 'The PyMOL Molecular Graphics System, Version 1.8'.</p>
<p>Tyanova, Stefka, Temu, Tikira, and Cox, Juergen (2016), 'The MaxQuant computational platform for mass spectrometry-based shotgun proteomics', Nature Protocols, 11 (12), 2301-19.</p>
<p>Yang, Bing, et al. (2012), 'Identification of cross-linked peptides from complex samples', Nature Methods, 9 (9), 904-06.</p>
<p>Zhang, Qi, et al. (2019), 'RNA exploits an exposed regulatory site to inhibit the enzymatic activity of PRC2', Nature Structural & Molecular Biology.</p>

	
	</div>
